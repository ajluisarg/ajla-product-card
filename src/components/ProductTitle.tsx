import React, { useContext } from "react";
import { ProductContext } from "./ProductCard";
import styles from "../styles/styles.module.css";

export const ProductTitle = ({title: providedTitle, className}: {title?: string, className?: string}) => {
    const {product: {title} = {title: ''}} = useContext(ProductContext);
  
    return <span className={`${styles.productDescription} ${className}`}>{providedTitle || title}</span>;
  };