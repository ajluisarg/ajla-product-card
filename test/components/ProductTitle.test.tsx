import React from 'react';
import renderer from 'react-test-renderer'
import {ProductCard, ProductTitle} from '../../src/components'
import { product1 } from './data/products';

describe('ProductTitle', () => {

    test('show component', () => {

        const wrapper = renderer.create(
            <ProductTitle title="Custom Product"/>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
        
    });

    test('show product title from context', () => {
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    () => (
                        <ProductTitle/>
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    })
});
