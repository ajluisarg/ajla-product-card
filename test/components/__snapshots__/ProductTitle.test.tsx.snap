// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle show component 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;

exports[`ProductTitle show product title from context 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee
  </span>
</div>
`;
